price-reporter/src/pair_info.rs

```diff
@@ -75,12 +75,20 @@ impl PairInfo {
             Token::from_addr_on_chain(parts[2], chain)
         };
 
-        Ok(Self::new(
-            exchange,
-            base.get_ticker().unwrap(),
-            quote.get_ticker().unwrap(),
-            Some(chain),
-        ))
+        let base_ticker = base.get_ticker().ok_or_else(|| {
+            ServerError::InvalidPairInfo(format!(
+                "{} and {} are not on the same chain",
+                base, quote
+            ))
+        })?;
+        let quote_ticker = quote.get_ticker().ok_or_else(|| {
+            ServerError::InvalidPairInfo(format!(
+                "{} and {} are not on the same chain",
+                base, quote
+            ))
+        })?;
```

Comment:
I would just make the error message `unable to get ticker for {}`. This could also happen if the token mapping is misconfigured or we request a price for some base/quote not in the mapping